Editor/Window/ManagedEC2Page.cs

```diff
@@ -0,0 +1,145 @@
+        private readonly Button _deployButton;
+        private readonly Button _redeployButton;
+        private readonly Button _deleteButton;
+        private readonly Button _launchClientButton;
+        private readonly DeploymentScenariosInput _deploymentScenariosInput;
+        private readonly FleetParametersInput _fleetParamsInput;
+        private readonly Label _ec2DeploymentStatusLabel;
+
+        public ManagedEC2Page(VisualElement container)
+        {
+            _container = container;
+            _model = DeploymentSettingsFactory.Create();
+            _model.Restore();
+            _model.Refresh();
+            var parameters = new FleetParameters
+            {
+                FleetName = _model.FleetName ?? $"{Application.productName}-ManagedFleet",
+                LaunchParameters = _model.LaunchParameters ?? $"",
+                BuildName = _model.BuildName ??
+                            $"{Application.productName}-{_model.ScenarioName.Replace(" ", "_")}-Build",
+                GameServerFile = _model.BuildFilePath,
+                GameServerFolder = _model.BuildFolderPath,
+                OperatingSystem = FleetParametersInput.GetOperatingSystem(_model.BuildOperatingSystem) ??
+                                  OperatingSystem.AMAZON_LINUX_2
+            };
+
+            var mVisualTreeAsset = UnityEngine.Resources.Load<VisualTreeAsset>("EditorWindow/Pages/ManagedEC2Page");
+            var uxml = mVisualTreeAsset.Instantiate();
+
+            container.Add(uxml);
+
+            var ec2Deployment = new ManagedEC2Deployment(_model, parameters);
+            var scenarioContainer = container.Q("ManagedEC2ScenarioTitle");
+            _deploymentScenariosInput =
+                new DeploymentScenariosInput(scenarioContainer,
+                    DeploymentScenariosInput.ScenarioIndexMap[_model.ScenarioIndex], true);
+            _deploymentScenariosInput.SetEnabled(true);
+            _deploymentScenariosInput.OnValueChanged += value => { Debug.Log($"Fleet type changed to {value}"); };
+            _ec2DeploymentStatusLabel = _container.Q<Label>("ManagedEC2DeployStatusText");
+
+            var parametersInput = container.Q<Foldout>("ManagedEC2ParametersTitle");
+            _fleetParamsInput = new FleetParametersInput(parametersInput, parameters);
+            _fleetParamsInput.OnValueChanged += param =>
+            {
+                ec2Deployment.UpdateModelFromParameters();
+                UpdateGUI();
+            };
+
+            _deployButton = container.Q<Button>("ManagedEC2CreateStackButton");
+            _deployButton.RegisterCallback<ClickEvent>(_ =>
+            {
+                ec2Deployment.StartDeployment();
+                UpdateGUI();
+            });
+            _redeployButton = container.Q<Button>("ManagedEC2RedeployStackButton");
+            _redeployButton.RegisterCallback<ClickEvent>(_ =>
+            {
+                ec2Deployment.StartDeployment();
```

Comment:
Is this functionality correct? If a stack is currently deploying, `StartDeployment` does nothing. What is the intended or expected functionality when this is clicked?

Reply:
It should get deployed ontop of the existing stack with an updated CF template/changeset